Samples/src/SimpleGrover/SimpleGrover.qs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+namespace Microsoft.Quantum.Samples.SimpleGrover {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Arithmetic;
+
+    // This operation adds a (-1)-phase to the marked state(s)
+    operation ReflectMarked (inputQubits : Qubit[]) : Unit is Adj + Ctl {
+        using (outputQubit = Qubit()) {
+            within {
+                // Initialize the outputQubit to 1/sqrt(2) ( |0> - |1> )
+                // so that toggling it results in a (-1)-phase
+                X(outputQubit);
+                H(outputQubit);
+            }
+            apply {
+                // Flip the outputQubit for marked states.
+                // Here: For the state with alternating 0s and 1s
+                within {
+                    ApplyToEachCA(X, inputQubits[0..2..Length(inputQubits)-1]);
```

Comment:
Considering to drop `Adj + Ctl` from this operation, this can become `ApplyToEach`

Reply:
Support for `Adjoint` is required for operations called inside a `within` block, but I agree that controlled shouldn't be required. It may be at the moment, though, as I'm not sure that https://github.com/microsoft/qsharp-compiler/pull/152 is included in the 0.9 release.